app/javascript/react/components/ThresholdConfigurator/ThresholdConfigurator.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import * as S from "./ThresholdConfigurator.style";
+
+const calculateThumbPercentage = (
+  value: number,
+  min: number,
+  max: number
+): number => {
+  const percentage = (value - min) / (max - min);
+  return percentage;
+};
+
+const calculateThumbPosition = (
+  value: number,
+  min: number,
+  max: number,
+  width: number
+): number => {
+  const percentage = calculateThumbPercentage(value, min, max);
+  return percentage * width;
+};
+
+interface ThresholdsConfiguratorProps {
+  initialThresholds: number[];
+}
+
+const ThresholdsConfigurator: React.FC<ThresholdsConfiguratorProps> = ({
+  initialThresholds,
+}) => {
+  const [thresholds, setThresholds] = useState(initialThresholds);
+  const [sliderWidth, setSliderWidth] = useState(0);
+  const [thumbPositions, setThumbPositions] = useState<number[]>([]);
+  const sliderRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (sliderRef.current) {
+      setSliderWidth(sliderRef.current.offsetWidth);
+    }
+    const [min, first, second, third, max] = thresholds;
```

Comment:
It's subjective but I like more `[min, low, middle, hight, max]` becouse now first is not exactly first

Reply:
Definitely!